tensorflow/compiler/xla/service/cpu/cpu_runtime_sse4_1.h

```diff
@@ -24,22 +24,33 @@ limitations under the License.
 
 #include "tensorflow/core/platform/macros.h"
 
+// MSVC does not have __SSE4_1__ macro. Eigen enables EIGEN_VECTORIZE_SSE4_1
+// when __AVX__ is defined, we should do the same.
+#if defined(__SSE4_1__) || defined(__AVX__)
+#include <smmintrin.h>
+#define TF_XLA_HAS_SSE4_1 1
+#else
+#define TF_XLA_HAS_SSE4_1 0
+#endif
+
```

Comment:
> Why do we need the new TF_XLA_HAS_* defines?

I use `__AVX__` to detect SSE4.1 for MSVC, so the macro to detect SSE4.1 becomes `defined(__SSE4_1__) || defined(__AVX__)`. Putting `defined(__SSE4_1__) || defined(__AVX__)` in every place might confuse reader since I only document the MSVC case in this place only. `TF_XLA_HAS_SSE4_1` provides clearer meaning.

tl;dr: For readability.

Reply:
Ok, that makes sense.  I had missed the fact that MSVC does not define  `__SSE4_1__`.